src/esp/gfx/CubeMapCamera.h

```diff
@@ -0,0 +1,125 @@
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_GFX_CUBEMAPCAMERA_H_
+#define ESP_GFX_CUBEMAPCAMERA_H_
+
+#include <Magnum/GL/CubeMapTexture.h>
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Tags.h>
+#include "esp/core/esp.h"
+#include "esp/gfx/RenderCamera.h"
+
+namespace esp {
+namespace gfx {
+class CubeMapCamera : public RenderCamera {
+ public:
+  /**
+   * @brief Constructor
+   * @param node, the scene node to which the camera is attached
+   */
+  explicit CubeMapCamera(scene::SceneNode& node);
+  /**
+   * @brief Constructor
+   * @param node, the scene node to which the camera is attached
+   * @param eye, the eye position (parent node space)
+   * @param target, the target position (parent node space)
+   * @param up, the up direction (parent node space)
+   */
+  explicit CubeMapCamera(scene::SceneNode& node,
+                         const vec3f& eye,
+                         const vec3f& target,
+                         const vec3f& up);
+  /**
+   * @brief Constructor
+   * @param node, the scene node to which the camera is attached
+   * @param eye, the eye position (parent node space)
+   * @param target, the target position (parent node space)
+   * @param up, the up direction (parent node space)
+   */
+  explicit CubeMapCamera(scene::SceneNode& node,
+                         const Magnum::Vector3& eye,
+                         const Magnum::Vector3& target,
+                         const Magnum::Vector3& up);
+  virtual ~CubeMapCamera(){};
+  /**
+   * @brief Move the camera towards a specified cube face
+   * @param cubeSide, the cube map coordinate, see the following pictures
+   * NOTE: +Y is top
+   *           +----+
+   *           | -Y |
+   * +----+----+----+----+
+   * | -Z | -X | +Z | +X |
+   * +----+----+----+----+
+   *           | +Y |
+   *           +----+
+   * CAREFUL! the local transformation of the camera node will be set after
+   * calling this function.
+   * @return Reference to self (for method chaining)
+   */
+  CubeMapCamera& switchToFace(Magnum::GL::CubeMapCoordinate cubeSide);
+  /**
+   * @brief Overload, move the camera towards a specified cube face
+   * @param cubSideIndex, the index of the cube map coordinate.
+   * 0: +X
+   * 1: -X
+   * 2: +Y
+   * 3: -Y
+   * 4: +Z
+   * 5: -Z
+   *           +----+
+   *           | -Y |
+   * +----+----+----+----+
+   * | -Z | -X | +Z | +X |
+   * +----+----+----+----+
+   *           | +Y |
+   *           +----+
+   * @return Reference to self (for method chaining)
+   */
+  CubeMapCamera& switchToFace(unsigned int cubeSideIndex);
+
+  /**
+   * Calling the the setProjectionMatrix from base class is not allowed
+   */
+  RenderCamera& setProjectionMatrix(int width,
+                                    int height,
+                                    float znear,
+                                    float zfar,
+                                    float hfov) = delete;
```

Comment:
> Calling the the setProjectionMatrix from base class is not allowed

Wait but the base implementation isn't `virtual` -- can it happen that someone calls this function from the base class pointer? And what about `setOrthographicProjectionMatrix()`, shouldn't it get deleted as well?

Alternatively and possibly more user friendly in case it's really intended to have those `virtual` and called from the base class pointer, would be to not delete this but instead assert that width and height is the same and FoV is 90°.

Reply:
My purpose to delete it is that any developer who accidentally calls this function from the base class would get caught by the compiler, so she can fix it immediately. ("fail-fast")

And the `RenderCamera` class is NOT the interface class of the `CubeMapCamera`. I did not add `virtual` because no polymorphism is needed here. Anyone who tries to use it in such a way is misinterpreting the purpose. Ideally, a `RenderCamera` pointer should never be pointing to a `CubeMapCamera` instance.

However, I also like the following suggestions though. But again, I do not think polymorphism is needed here.
>Alternatively and possibly more user friendly in case it's really intended to have those virtual and called from the base class pointer, would be to not delete this but instead assert that width and height is the same and FoV is 90°.

Reply:
> CubeMapCamera is subclassed from RenderCamera simply to avoid duplicating functions. Ideally, a RenderCamera pointer should never be pointing to a CubeMapCamera instance.

I got some crazy ideas about private inheritance but ... eh. Keep it like this, then :)